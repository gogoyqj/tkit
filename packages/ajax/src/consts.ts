/**
 * @author: yangqianjun
 * @file: ajax 常量
 * @Date: 2019-12-06 18:32:12
 * @LastEditors: yangqianjun
 * @LastEditTime: 2020-01-08 18:31:51
 */

import { TkitUtils } from 'tkit-types';

/** 取消请求错误码 */
export const AjaxCancelCode = 10499;
export const AjaxCancelMessage = 'AJAX_CANCEL_MESSAGE';

/** 默认Ajax错误码  */
export const AjaxErrorCode = 10001;

/** 无法确认的 Graphql 错误码 */
export const GraphQLErrorCode = 10002;

export interface TkitAbstractAjaxResult<R> {
  /**
   * 后端返回的错误码或者提取非200的http状态码
   */
  code?: number;
  /**
   * 后端返回的错误信息
   */
  message?: string | number | null;
  /**
   * 后端返回的数据结构
   */
  result?: R;
}
export type TkitAjaxResult = TkitAbstractAjaxResult<any>;
export type TkitAjaxFunction = TkitUtils.AbstractAsyncFunction<TkitAjaxResult>;
export const promiseFactory = <T>() => {
  let resolve!: (value?: T | PromiseLike<T> | undefined) => void, reject!: (reason?: any) => void;
  const prom = new Promise<T>((rs, rj) => {
    resolve = rs;
    reject = rj;
  });
  const rj = { reject, resolve };
  return [rj, prom] as [typeof rj, typeof prom];
};
