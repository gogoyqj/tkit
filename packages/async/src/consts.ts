/**
 * @author: yangqianjun
 * @file: async常量
 * @Date: 2019-09-11 11:02:36
 * @LastEditors: yangqianjun
 * @LastEditTime: 2019-12-06 18:37:18
 */
import { AsyncModelType } from './asyncModel';

export const ASYNC_EFFECT_EVENT_NAME = 'ASYNC_EFFECT_EVENT_NAME';
export interface AsyncEffectEventType {
  type: keyof AsyncModelType['actions'];
  payload: any;
}

export const ASYNC_RESULT_EVENT_NAME = 'ASYNC_RESULT_EVENT_NAME';
export interface AsyncResultEventType {
  /**
   * 副作用信息类型
   */
  type: 'error' | 'success';
  /**
   * 副作用提示信息
   */
  message: React.ReactNode;
}

/** 异步队列变化事件 */
export const ASYNC_STATUS_CHANGE_NAME = 'ASYNC_STATUS_CHANGE_NAME';
