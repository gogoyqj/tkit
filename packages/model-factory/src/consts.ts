import { EffectOptions } from './types';

/**
 * @file: 常量
 * @author: yangqianjun
 * @Date: 2020-02-06 16:27:23
 * @LastEditors: yangqianjun
 * @LastEditTime: 2020-02-06 17:30:29
 */

/** 默认副作用配置 */
export const defaultEffectOptions: EffectOptions = {
  type: 'takeEvery',
  ms: 100
};

/** 如引入 immer，枚举标记 redux 或者 react hooks */
export const enum ModernType {
  /** redux */
  ReduxModern = 0,
  /** useModel */
  HookModern = 1
}

/**
 * @file: model 常量
 * @author: yangqianjun
 * @Date: 2020-02-06 14:42:35
 * @LastEditors: yangqianjun
 * @LastEditTime: 2020-02-06 14:52:14
 */

/**
 * 执行 effect 广播的事件
 * 从 tkit-async 拆过来，以逐步解除 model & async 的循环引用
 */
export const ASYNC_EFFECT_EVENT_NAME = 'ASYNC_EFFECT_EVENT_NAME';

/** Redux Type 副作用开始 */
export const EFFECTS_START = 'EFFECTS_START';

/** Redux Type 副作用结束 */
export const EFFECTS_END = 'EFFECTS_END';

/** Redux Type 副作用错误 */
export const EFFECTS_ERROR = 'EFFECTS_ERROR';

/** Redux Type 副作用成功 */
export const EFFECTS_SUCCESS = 'EFFECTS_SUCCESS';

/**  副作用消息EventCenter事件 */
export const EFFECTS_MSG = 'EFFECTS_MSG';

export const fakeEffectRes = { code: 0 };
