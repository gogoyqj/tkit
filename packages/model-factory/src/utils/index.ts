/**
 * @file: description
 * @author: yangqianjun
 * @Date: 2020-02-06 16:33:09
 * @LastEditors: yangqianjun
 * @LastEditTime: 2020-02-07 15:40:25
 */

export * from './putWrapper';
export * from './async';
