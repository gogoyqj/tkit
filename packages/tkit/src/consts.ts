import path from 'path';
import fs from 'fs';
import { VariableDeclarationKind } from 'ts-morph';

export const CONST = 'const' as VariableDeclarationKind;
export const OPTIONS = {
  async: ['-a, --async', 'auto async in features/xxx/index.tsx'],
  connect: ['-c, --connect', 'connect with redux'],
  event: ['-e, --event', 'wrapp with EventWrapper'],
  force: ['-f, --force', 'overlap if file already exists'],
  hooks: ['-k, --hooks', 'import react hooks in presenter'],
  mobile: [
    '-m, --mobile [mobile]',
    'create Component or Presenter like .h5.ts, if set web, then .web.ts[only if default is mobile]'
  ],
  pure: ['-p, --pure', 'use React.PureComponent insted of React.Component'],
  url: ['-u, --url', "create 'NamePage' like Component"],
  wrap: [
    '-w, --wrap [wrap]',
    "wrap 'create component or presenter' in a sub directory, default: true",
    true
  ],
  local: [
    '-l, --local',
    "create localModel that won't connect with global store, default: false",
    false
  ]
};
export const coreDirectory = path.join(__dirname, '..');
export const templatesDir = path.join(coreDirectory, 'templates');
export const appDirectory = fs.realpathSync(process.cwd());
export const baseSourceDir = path.join(appDirectory, 'src');
export const baseTestsDir = path.join(appDirectory, '__tests__');
