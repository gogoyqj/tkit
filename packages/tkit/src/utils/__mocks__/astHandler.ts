export const astHandler = jest.fn();
