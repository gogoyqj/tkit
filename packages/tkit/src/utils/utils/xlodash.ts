/**
 * @description lodash 扩展
 */
import _ from 'lodash';

export default {
  ..._,
  pascalCase: _.flow(_.camelCase, _.upperFirst) as (s: string) => string,
  upperSnakeCase: _.flow(_.snakeCase, _.toUpper) as (s: string) => string
};
