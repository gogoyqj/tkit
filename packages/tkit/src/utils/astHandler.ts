/**
 * @description ts ast 操作封装
 */
import path from 'path';
import {
  Project,
  ProjectOptions,
  IndentationText,
  QuoteKind,
  VariableDeclarationKind,
  SourceFile,
  ImportDeclarationStructure
} from 'ts-morph';
import { appDirectory } from '../consts';

export const defaultIndentationText = '  ';

export function defaultCodeWriter(
  type: string,
  bodyArr: string[],
  indentationText = defaultIndentationText,
  start = '',
  end = ''
) {
  switch (type) {
    default:
      return `${start ? `${start}\n` : ''}${
        bodyArr.length ? `${indentationText}${bodyArr.join(`,\n${indentationText}`)}` : ''
      }${end ? `\n${end}` : ''}`;
  }
}

export type OperationType = 'remove' | 'add';
export interface OperationConfig {
  moduleSpecifier?: string;
  vars?: {
    declarationKind: VariableDeclarationKind;
    declarations: {
      name: string;
      initializer: string;
    }[];
    codeWriter?: typeof defaultCodeWriter;
  }[];
  namedExports?: string | string[];
  namedImports?: string | string[];
  defaultImport?: string;
}
export interface OperationFileMap {
  [filePath: string]: OperationConfig;
}

export function createProject(config?: ProjectOptions) {
  return new Project({
    manipulationSettings: {
      quoteKind: "'" as QuoteKind,
      indentationText: '  ' as IndentationText
    },
    tsConfigFilePath: path.join(appDirectory, 'tsconfig.json'),
    ...config
  });
}

export async function astHandler(
  fileMapToInject: OperationFileMap,
  action: OperationType = 'add',
  config?: ProjectOptions
) {
  console.time('parse');
  const isRemove = action === 'remove';
  const project = createProject(config);
  Object.keys(fileMapToInject).forEach(file => {
    // const root = project.addExistingSourceFile(file, fs.readFileSync(file, { encoding: 'utf8' }));
    const root = project.addExistingSourceFile(file);
    const config = fileMapToInject[file];
    if (typeof config !== 'object' || Array.isArray(config)) {
      return console.error(`Error: ${file} with invalid config`);
    }
    if (config.namedImports) {
      if (!Array.isArray(config.namedImports)) {
        config.namedImports = [config.namedImports];
      }
    }
    if (config.vars) {
      if (!Array.isArray(config.vars)) {
        config.vars = [config.vars];
      }
    }
    if (config.namedExports) {
      if (!Array.isArray(config.namedExports)) {
        config.namedExports = [config.namedExports];
      }
    }
    handleVars(root, config, isRemove);
    handleImport(root, config, isRemove);
    // hanleExport(root, config, isRemove);
  });
  await project.save();
  console.timeEnd('parse');
}

export function handleVars(
  root: SourceFile,
  config: Pick<OperationConfig, 'vars'>,
  isRemove: boolean = false
) {
  if (config.vars) {
    const { vars } = config;
    vars.forEach(v => {
      const { declarationKind, declarations, codeWriter = defaultCodeWriter } = v;
      if (Array.isArray(declarations)) {
        let newDeclarations = declarations;
        root.getVariableStatement(decl => {
          const _declarationKind = decl.getDeclarationKind();
          const namesNode = decl.getDeclarations();
          if (_declarationKind === declarationKind) {
            namesNode.forEach(nameNode => {
              const nameString = nameNode.getName();
              newDeclarations = newDeclarations.filter(declaration => {
                let { name, initializer } = declaration;
                if (nameString === name) {
                  const initializerNodes = nameNode.getInitializer();
                  if (initializerNodes) {
                    const arrStartNode = initializerNodes.getFirstChild();
                    let arrStart = arrStartNode && arrStartNode.getText();
                    const arrEndNode = initializerNodes.getLastChild();
                    let arrEnd = arrEndNode && arrEndNode.getText();
                    const initializers = [];
                    let exitst;
                    if (
                      (arrStart === '[' || arrStart === '{') &&
                      (arrEnd === ']' || arrEnd === '}')
                    ) {
                      initializerNodes.forEachChild(initializerNode => {
                        const initializerString = initializerNode.getText();
                        if (initializerString === initializer) {
                          exitst = true;
                          if (!isRemove) {
                            initializers.push(initializerString);
                          }
                        } else {
                          initializers.push(initializerString);
                        }
                      });
                    } else {
                      // 非数组、对象：替换
                      arrStart = arrEnd = '';
                    }
                    if (!exitst) {
                      initializers.push(initializer);
                    }
                    nameNode.setInitializer(
                      codeWriter(
                        'initializer',
                        initializers,
                        defaultIndentationText,
                        arrStart,
                        arrEnd
                      )
                    );
                  }
                  return false;
                }
                return true;
              });
            });
          }
          return false;
        });
        if (newDeclarations.length) {
          const statement = root.addVariableStatement({
            declarationKind,
            declarations: newDeclarations
          });
          statement.setIsExported(true);
        }
      }
    });
  }
}

export function handleImport(
  root: SourceFile,
  config: Pick<OperationConfig, 'defaultImport' | 'namedImports' | 'moduleSpecifier'>,
  isRemove: boolean = false
) {
  let { moduleSpecifier, namedImports } = config;
  const importDeclaration = root.getImportDeclaration(dec => {
    return dec.getModuleSpecifier().getText() === `'${moduleSpecifier}'`;
  });
  if (importDeclaration) {
    // update or remove
    let oldNamedImports = (importDeclaration.getNamedImports() || []).map(a => a.getText());
    if (isRemove) {
      importDeclaration.remove();
    } else {
      if (config.defaultImport) {
        importDeclaration.setDefaultImport(config.defaultImport);
      }
      if ('defaultImport' in config) {
        importDeclaration.remove();
      }
      if (namedImports) {
        (Array.isArray(namedImports) ? namedImports : [namedImports]).forEach(newItem => {
          if (oldNamedImports.indexOf(newItem) === -1) {
            oldNamedImports.push(newItem);
          }
        });
        importDeclaration.removeNamedImports();
        importDeclaration.addNamedImports(oldNamedImports);
      } else if ('namedImports' in config) {
        importDeclaration.removeNamedImports();
      }
    }
  } else if (moduleSpecifier && !isRemove) {
    root.addImportDeclaration(config as ImportDeclarationStructure);
  }
}

/**
 * @description @todo
 */
export function handleDefaultExport(
  root: SourceFile,
  config: Pick<OperationConfig, 'namedExports'>,
  isRemove: boolean = false
) {
  const defaultExport = root.getDefaultExportSymbol();
}

/**
 * @description @todo
 */
export function hanleExport(root: SourceFile, config: OperationConfig, isRemove: boolean = false) {
  let { namedExports } = config;
  // const exportDeclaration = root.getExportedDeclarations(dec => {
  //   return dec.getModuleSpecifier().getText() === `'${moduleSpecifier}'`;
  // }
  const exportDeclaration = root.getExportedDeclarations();
  if (namedExports) {
    if (exportDeclaration) {
      const newAdd = [];
      exportDeclaration.forEach(declaration => {
        console.log(declaration);
      });
    }
  }
}
