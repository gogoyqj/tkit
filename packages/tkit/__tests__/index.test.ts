import path from 'path';
import fsX from 'fs-extra';
import { ProjectOptions } from 'ts-morph';

import { action } from '@src/index';
import { astHandler, OperationFileMap, OperationType } from '@src/utils/astHandler';
import { coreDirectory } from '@src/consts';

const featuresDirectory = path.join(coreDirectory, 'src', 'features');
const testsDirectory = path.join(coreDirectory, 'tests');

function clearBoilerplate() {
  fsX.removeSync(featuresDirectory);
  fsX.removeSync(testsDirectory);
}

// @cc: 必须置于顶层
jest.mock('@src/utils/astHandler');

const { clearASTLogs, ASTLogs } = (() => {
  const ASTLogs: any[] = [];
  (astHandler as jest.Mock).mockImplementation(
    (fileMapToInject: OperationFileMap, action: OperationType, config?: ProjectOptions) => {
      ASTLogs.push([
        Object.keys(fileMapToInject).reduce((o, file) => {
          const relative = file.split(coreDirectory)[1];
          o[relative] = fileMapToInject[file];
          return o;
        }, {}),
        action,
        config
      ]);
    }
  );
  return { clearASTLogs: () => ASTLogs.splice(0, ASTLogs.length), ASTLogs };
})();

// @cc: 必须置于顶层
// const writeFileSync = jest.spyOn(fsX, 'writeFileSync');
const { WriteLogs, clearWriteLogs, writeFileSync } = (() => {
  const WriteLogs: any[] = [];

  const writeFileSync = (fsX.writeFileSync = jest.fn(
    (file: string, content: string, config: any) => {
      // @cc: 只校验文件名是否符合预期
      WriteLogs.push([file.split(coreDirectory)[1]]);
    }
  ));

  return {
    clearWriteLogs: () => WriteLogs.splice(0, WriteLogs.length),
    WriteLogs,
    writeFileSync
  };
})();

describe('tkit-core', () => {
  afterEach(async () => {
    clearASTLogs();
    clearWriteLogs();
    await new Promise(rs => setTimeout(rs, 100));
  });
  beforeAll(() => {
    clearBoilerplate();
  });

  afterAll(() => {
    clearBoilerplate();
  });

  it('should add Component/init works ok', () => {
    clearBoilerplate();
    action('Component', 'admin/home', { force: true });
    expect(astHandler).toBeCalled();
    expect(ASTLogs.length).toBeGreaterThan(0);
    expect(ASTLogs).toMatchSnapshot();

    expect(writeFileSync).toBeCalled();
    expect(WriteLogs.length).toBeGreaterThan(0);
    expect(WriteLogs).toMatchSnapshot();
  });

  it('should add Component/directory/no-init works ok', () => {
    action('Component', 'admin/directory/home', { force: true });
    expect(astHandler).toBeCalled();
    // expect(ASTLogs.length).toBeGreaterThan(0);
    expect(ASTLogs).toMatchSnapshot();

    expect(writeFileSync).toBeCalled();
    expect(WriteLogs.length).toBeGreaterThan(0);
    expect(WriteLogs).toMatchSnapshot();
  });

  it('should add Component/no-init/wrap works ok', () => {
    action('Component', 'admin/home', { force: true, wrap: true, url: true });
    expect(astHandler).toBeCalled();
    // expect(ASTLogs.length).toBeGreaterThan(0);
    expect(ASTLogs).toMatchSnapshot();

    expect(writeFileSync).toBeCalled();
    expect(WriteLogs.length).toBeGreaterThan(0);
    expect(WriteLogs).toMatchSnapshot();
  });

  it('should add Component/no-init/url works ok', () => {
    action('Component', 'admin/home', { force: true, wrap: true, url: true });
    expect(astHandler).toBeCalled();
    // expect(ASTLogs.length).toBeGreaterThan(0);
    expect(ASTLogs).toMatchSnapshot();

    expect(writeFileSync).toBeCalled();
    expect(WriteLogs.length).toBeGreaterThan(0);
    expect(WriteLogs).toMatchSnapshot();
  });

  it('should add Presenter/no-init works ok', () => {
    action('Presenter', 'admin/home', { force: true });
    expect(astHandler).toBeCalled();
    // expect(ASTLogs.length).toBeGreaterThan(0);
    expect(ASTLogs).toMatchSnapshot();

    expect(writeFileSync).toBeCalled();
    expect(WriteLogs.length).toBeGreaterThan(0);
    expect(WriteLogs).toMatchSnapshot();
  });

  it('shoul add Model/no-init works ok', () => {
    action('Model', 'admin/userModel', { force: true });
    expect(astHandler).toBeCalled();
    expect(ASTLogs.length).toBeGreaterThan(0);
    expect(ASTLogs).toMatchSnapshot();

    expect(writeFileSync).toBeCalled();
    expect(WriteLogs.length).toBeGreaterThan(0);
    expect(WriteLogs).toMatchSnapshot();
  });

  it('shoul add List/no-init works ok', () => {
    action('List', 'admin/userList', { force: true });
    expect(astHandler).toBeCalled();
    expect(ASTLogs.length).toBeGreaterThan(0);
    expect(ASTLogs).toMatchSnapshot();

    expect(writeFileSync).toBeCalled();
    expect(WriteLogs.length).toBeGreaterThan(0);
    expect(WriteLogs).toMatchSnapshot();
  });

  it('shoul add Action/no-init works ok', () => {
    action('Action', 'admin/userInfo', { force: true });
    expect(astHandler).toBeCalled();
    expect(ASTLogs.length).toBeGreaterThan(0);
    expect(ASTLogs).toMatchSnapshot();

    expect(writeFileSync).toBeCalled();
    expect(WriteLogs.length).toBeGreaterThan(0);
    expect(WriteLogs).toMatchSnapshot();
  });

  it('shoul add Selector/no-init works ok', () => {
    action('Selector', 'admin/selectUser', { force: true });
    expect(astHandler).toBeCalled();
    // expect(ASTLogs.length).toBeGreaterThan(0);
    expect(ASTLogs).toMatchSnapshot();

    expect(writeFileSync).toBeCalled();
    expect(WriteLogs.length).toBeGreaterThan(0);
    expect(WriteLogs).toMatchSnapshot();
  });
});
