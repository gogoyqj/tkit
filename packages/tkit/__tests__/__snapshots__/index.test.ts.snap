// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tkit-core shoul add Action/no-init works ok 1`] = `
Array [
  Array [
    Object {
      "/src/features/admin/redux/actions.ts": Object {
        "moduleSpecifier": "./userInfo",
        "namedImports": "doUserInfo",
        "vars": Array [
          Object {
            "declarationKind": "const",
            "declarations": Array [
              Object {
                "initializer": "doUserInfo",
                "name": "actions",
              },
            ],
          },
        ],
      },
      "/src/features/admin/redux/initialState.ts": Object {
        "moduleSpecifier": "./userInfo",
        "namedImports": "pureUserInfoState",
        "vars": Array [
          Object {
            "declarationKind": "const",
            "declarations": Array [
              Object {
                "initializer": "...pureUserInfoState",
                "name": "initialState",
              },
            ],
          },
        ],
      },
      "/src/features/admin/redux/reducer.ts": Object {
        "moduleSpecifier": "./userInfo",
        "namedImports": "applyUserInfo",
        "vars": Array [
          Object {
            "declarationKind": "const",
            "declarations": Array [
              Object {
                "initializer": "applyUserInfo",
                "name": "reducers",
              },
            ],
          },
        ],
      },
    },
    undefined,
    undefined,
  ],
]
`;

exports[`tkit-core shoul add Action/no-init works ok 2`] = `
Array [
  Array [
    "/src/features/admin/redux/userInfo.ts",
  ],
  Array [
    "/__tests__/features/admin/redux/userInfo.test.ts",
  ],
]
`;

exports[`tkit-core shoul add List/no-init works ok 1`] = `
Array [
  Array [
    Object {
      "/src/features/admin/redux/actions.ts": Object {
        "moduleSpecifier": "./userList",
        "namedImports": "doUserList, doSelectUserList",
        "vars": Array [
          Object {
            "declarationKind": "const",
            "declarations": Array [
              Object {
                "initializer": "doUserList",
                "name": "actions",
              },
              Object {
                "initializer": "doSelectUserList",
                "name": "actions",
              },
            ],
          },
        ],
      },
      "/src/features/admin/redux/initialState.ts": Object {
        "moduleSpecifier": "./userList",
        "namedImports": "pureUserListState",
        "vars": Array [
          Object {
            "declarationKind": "const",
            "declarations": Array [
              Object {
                "initializer": "...pureUserListState",
                "name": "initialState",
              },
            ],
          },
        ],
      },
      "/src/features/admin/redux/reducer.ts": Object {
        "moduleSpecifier": "./userList",
        "namedImports": "applyUserList",
        "vars": Array [
          Object {
            "declarationKind": "const",
            "declarations": Array [
              Object {
                "initializer": "applyUserList",
                "name": "reducers",
              },
            ],
          },
        ],
      },
    },
    undefined,
    undefined,
  ],
]
`;

exports[`tkit-core shoul add List/no-init works ok 2`] = `
Array [
  Array [
    "/src/features/admin/redux/userList.ts",
  ],
  Array [
    "/src/features/admin/redux/sagas.ts",
  ],
  Array [
    "/__tests__/features/admin/redux/userList.test.ts",
  ],
  Array [
    "/src/features/admin/redux/sagas.ts",
  ],
]
`;

exports[`tkit-core shoul add Model/no-init works ok 1`] = `
Array [
  Array [
    Object {
      "/src/features/admin/redux/actions.ts": Object {
        "moduleSpecifier": "./userModel",
        "namedImports": "default as doUserModel",
        "vars": Array [
          Object {
            "declarationKind": "const",
            "declarations": Array [
              Object {
                "initializer": "...doUserModel.actions",
                "name": "actions",
              },
            ],
          },
        ],
      },
      "/src/features/admin/redux/initialState.ts": Object {
        "moduleSpecifier": "./userModel",
        "namedImports": "userModelState",
        "vars": Array [
          Object {
            "declarationKind": "const",
            "declarations": Array [
              Object {
                "initializer": "...userModelState",
                "name": "initialState",
              },
            ],
          },
        ],
      },
      "/src/features/admin/redux/reducer.ts": Object {
        "moduleSpecifier": "./userModel",
        "namedImports": "default as doUserModel",
        "vars": Array [
          Object {
            "declarationKind": "const",
            "declarations": Array [
              Object {
                "initializer": "doUserModel.reducers",
                "name": "reducers",
              },
            ],
          },
        ],
      },
      "/src/features/admin/redux/sagas.ts": Object {
        "moduleSpecifier": "./userModel",
        "namedImports": "default as doUserModel",
        "vars": Array [
          Object {
            "declarationKind": "const",
            "declarations": Array [
              Object {
                "initializer": "doUserModel.sagas",
                "name": "sagaUserModel",
              },
            ],
          },
        ],
      },
    },
    undefined,
    undefined,
  ],
]
`;

exports[`tkit-core shoul add Model/no-init works ok 2`] = `
Array [
  Array [
    "/src/features/admin/redux/userModel.ts",
  ],
]
`;

exports[`tkit-core shoul add Selector/no-init works ok 1`] = `Array []`;

exports[`tkit-core shoul add Selector/no-init works ok 2`] = `
Array [
  Array [
    "/src/features/admin/selectors/selectUser.ts",
  ],
  Array [
    "/src/features/admin/selectors/index.ts",
  ],
  Array [
    "/__tests__/features/admin/selectors/selectUser.test.ts",
  ],
  Array [
    "/src/features/admin/selectors/index.ts",
  ],
]
`;

exports[`tkit-core should add Component/directory/no-init works ok 1`] = `Array []`;

exports[`tkit-core should add Component/directory/no-init works ok 2`] = `
Array [
  Array [
    "/src/features/admin/directoryComponents/Home.tsx",
  ],
  Array [
    "/src/features/admin/directoryComponents/index.ts",
  ],
  Array [
    "/__tests__/features/admin/directoryComponents/Home.test.tsx",
  ],
  Array [
    "/src/features/admin/directoryComponents/index.ts",
  ],
  Array [
    "/src/features/admin/directoryComponents/Home.less",
  ],
  Array [
    "/src/features/admin/directoryComponents/index.ts",
  ],
]
`;

exports[`tkit-core should add Component/init works ok 1`] = `
Array [
  Array [
    Object {
      "/src/common/rootReducer.ts": Object {
        "defaultImport": "adminReducer",
        "moduleSpecifier": "@features/admin/redux/reducer",
        "vars": Array [
          Object {
            "declarationKind": "const",
            "declarations": Array [
              Object {
                "initializer": "admin: adminReducer",
                "name": "reducerMap",
              },
            ],
          },
        ],
      },
      "/src/common/rootSaga.ts": Object {
        "defaultImport": "* as adminSaga",
        "moduleSpecifier": "@features/admin/redux/sagas",
        "vars": Array [
          Object {
            "declarationKind": "const",
            "declarations": Array [
              Object {
                "initializer": "adminSaga",
                "name": "featureSagas",
              },
            ],
          },
        ],
      },
      "/src/common/routeConfig.ts": Object {
        "defaultImport": "adminRoute",
        "moduleSpecifier": "@features/admin/route",
        "vars": Array [
          Object {
            "declarationKind": "const",
            "declarations": Array [
              Object {
                "initializer": "adminRoute",
                "name": "childRoutes",
              },
            ],
          },
        ],
      },
    },
    undefined,
    undefined,
  ],
]
`;

exports[`tkit-core should add Component/init works ok 2`] = `
Array [
  Array [
    "/src/features/admin/route.ts",
  ],
  Array [
    "/src/features/admin/index.ts",
  ],
  Array [
    "/src/features/admin/Home.tsx",
  ],
  Array [
    "/src/features/admin/index.ts",
  ],
  Array [
    "/__tests__/features/admin/Home.test.tsx",
  ],
  Array [
    "/src/features/admin/index.ts",
  ],
  Array [
    "/src/features/admin/Home.less",
  ],
  Array [
    "/src/features/admin/index.ts",
  ],
]
`;

exports[`tkit-core should add Component/no-init/url works ok 1`] = `
Array [
  Array [
    Object {
      "/src/features/admin/route.ts": Object {
        "vars": Array [
          Object {
            "declarationKind": "const",
            "declarations": Array [
              Object {
                "initializer": "{
    load: loader('HomePage'),
    path: 'home'
  }",
                "name": "childRoutes",
              },
            ],
          },
        ],
      },
    },
    undefined,
    undefined,
  ],
]
`;

exports[`tkit-core should add Component/no-init/url works ok 2`] = `
Array [
  Array [
    "/src/features/admin/HomePage/HomePage.tsx",
  ],
  Array [
    "/src/features/admin/index.ts",
  ],
  Array [
    "/__tests__/features/admin/HomePage.test.tsx",
  ],
  Array [
    "/src/features/admin/index.ts",
  ],
  Array [
    "/src/features/admin/HomePage/HomePage.less",
  ],
  Array [
    "/src/features/admin/HomePage/index.ts",
  ],
]
`;

exports[`tkit-core should add Component/no-init/wrap works ok 1`] = `
Array [
  Array [
    Object {
      "/src/features/admin/route.ts": Object {
        "vars": Array [
          Object {
            "declarationKind": "const",
            "declarations": Array [
              Object {
                "initializer": "{
    load: loader('HomePage'),
    path: 'home'
  }",
                "name": "childRoutes",
              },
            ],
          },
        ],
      },
    },
    undefined,
    undefined,
  ],
]
`;

exports[`tkit-core should add Component/no-init/wrap works ok 2`] = `
Array [
  Array [
    "/src/features/admin/HomePage/HomePage.tsx",
  ],
  Array [
    "/src/features/admin/index.ts",
  ],
  Array [
    "/__tests__/features/admin/HomePage.test.tsx",
  ],
  Array [
    "/src/features/admin/index.ts",
  ],
  Array [
    "/src/features/admin/HomePage/HomePage.less",
  ],
  Array [
    "/src/features/admin/HomePage/index.ts",
  ],
]
`;

exports[`tkit-core should add Presenter/no-init works ok 1`] = `Array []`;

exports[`tkit-core should add Presenter/no-init works ok 2`] = `
Array [
  Array [
    "/src/features/admin/Home.tsx",
  ],
  Array [
    "/src/features/admin/index.ts",
  ],
  Array [
    "/__tests__/features/admin/Home.test.tsx",
  ],
  Array [
    "/src/features/admin/index.ts",
  ],
  Array [
    "/src/features/admin/Home.less",
  ],
  Array [
    "/src/features/admin/index.ts",
  ],
]
`;
